pages/verify/[discordServerId]/[discordMemberId]/[customLink].tsx

```diff
@@ -107,25 +168,37 @@ const VerifyPage = ({ discordServerName, starknetNetwork }: Props) => {
   const sign = useCallback(async () => {
     if (!account) return;
     try {
-      const messageCopy = {
-        ...messageToSign,
-        domain: { ...messageToSign.domain, chainId },
-      };
-      const signature = await account.signMessage(messageCopy);
-      await verifySignature(signature);
+      let signature;
+      if ("publicKey" in account) {
+        const message = "Your message to sign"; // Define your message here
+        signature = await stellarSignMessage(message, {
+          address: account.publicKey,
+        });
+      } else {
+        const messageCopy = {
+          ...messageToSign,
+          domain: { ...messageToSign.domain, chainId },
+        };
+        signature = messageCopy;
+      }
     } catch (e: any) {
       WatchTowerLogger.error(e.message, e);
     }
   }, [account, verifySignature, chainId]);
```

Comment:
React Hook useCallback has an unnecessary dependency: 'verifySignature'. Either exclude it or remove the dependency array.